model/project_ref_test.go

```diff
@@ -1253,6 +1253,11 @@ func TestDetachFromRepo(t *testing.T) {
 			_, err := pVars.Upsert()
 			assert.NoError(t, err)
 
+			dbProjRef, err := FindBranchProjectRef(pRef.Id)
```

Comment:
Can we add a check to make sure that it's not synced before the upsert and the upsert is what syncs it?

Reply:
That should already be true because the project ref defined just above this doesn't set it. I don't think it's really worth testing the initial state that explicitly since it's right there and this check will go away after the migration is complete anyways.